docs/src/preprocessing/preprocessing.md

```diff
@@ -1,6 +1,224 @@
+
+#### Continuous geometry
+
+For an arbitrary continuous geometry, the steps are as follows.
+
+```@raw html
+<figure>
+  <img src="https://github.com/user-attachments/assets/66c423ae-28ce-4d48-aa52-777726a3a677" alt="continuous closing"/>
+</figure>
+```
+
+(1):
+- Recurse left: $w_{\text{left}} = \text{\texttt{hierarchical\_winding}} (\mathbf{p}, T.\text{left})$
+- Recurse right: $w_{\text{right}} = \text{\texttt{hierarchical\_winding}} (\mathbf{p},T.\text{right})$
+
+(2):
+- Query point $\mathbf{p}$ is outside bounding box $T$.
+- Use closure $T.\bar{\mathcal{S}}$.
+- Note the negative winding $\rightarrow w_{\mathcal{S}}(\mathbf{p}) = - w_{\bar{\mathcal{S}}}(\mathbf{p})$
+
+$w_{\text{left}} = -\text{\texttt{naive\_winding}} (\mathbf{p}, T.\bar{\mathcal{S}})$
+
+(3):
+- Bounding box $T$ is a leaf. Use open surface $T.\mathcal{S}$
+
+$w_{\text{right}} = \text{\texttt{naive\_winding}} (\mathbf{p}, T.\mathcal{S})$
+
+The reconstructed surface will then look as in the following image.
+
+```@raw html
+<figure>
+  <img src="https://github.com/user-attachments/assets/920bb4f1-1336-4e77-b06d-d5b46ca0d8d5" alt="reconstructed surface"/>
+</figure>
+```
+
+We finally sum up the winding numbers
+
+$w = w_{\text{left}} + w_{\text{right} } = -w_{T_{\text{left}}.\bar{\mathcal{S}}} + w_{T_{\text{right}}.\mathcal{S}}$
+
+#### Discrete geometry
+
+```@raw html
+<figure>
+  <img src="https://github.com/user-attachments/assets/a9b59cc3-5421-40af-b0b0-f4c18a5a7078" alt="discrete geometry"/>
+</figure>
+```
+
+To construct the hierarchy for discrete piecewise-linear surfaces (1), we have to do the following.
+
+(2):
+Each edge is distributed to the child whose box contains the edge's barycenter.
+Splitting stops when the number of a box's edges slips below a
+threshold (usually $\approx 100$ faces in 3D, here: 6 edges).
+
+(3):
+For the closure, Jacobson et al. (2013) define *exterior vertices* (*exterior edges* in 3D)
+as boundary vertices of such a segmentation.
+To find them, we traverse around each edge (face in 3D) in order, and
+increment or decrement for each vertex (edge) a specific counter.
+
+```julia
+  # 2D
+  v1 = edge_vertices_ids[edge][1]
+  v2 = edge_vertices_ids[edge][2]
+
+  vertex_count[v1] += 1
+  vertex_count[v2] -= 1
+```
+```julia
+  # 3D
+  edge1 = face_edges_ids[face][1]
+
+  if edge_vertices_ids[edge1] == (v1, v2)
+      directed_edges[edge1] += 1
+  else
+      directed_edges[edge1] -= 1
+  end
+```
+
+In 2D, a vertex is declared as exterior if `vertex_count(vertex) != 0`.
+The exterior vertices are then connected to one arbitrary
+exterior vertex using appropriately oriented line segments
+
+```julia
+  edge = vertex_count[v] > 0 ? (closing_vertex, v) : (v, closing_vertex)
+```
+
+In 3D, an edge is declared as exterior if `directed_edges[edge] != 0`
+The exterior edges are triangulated with one exterior vertex $k$ with orientation
+- ``\{i, j, k\}`` if `directed_edges[edge] < 0`
+- ``\{j, i, k\}`` if `directed_edges[edge] > 0`
```

Comment:
````suggestion
In 2D, a vertex is declared as exterior if `vertex_count(vertex) != 0`, so there is not the same amount of edges in this box going into vs out of the vertex.
To construct the closing surface, the exterior vertices are then connected to one arbitrary
exterior vertex using appropriately oriented line segments

```julia
edge = vertex_count[v] > 0 ? (closing_vertex, v) : (v, closing_vertex)
```
The resulting closed surface ``T.S \cup T.\bar{S}`` then has the same number of edges going into and out of each vertex.
````
Again, I wouldn't show the 3D case.